src/rar2fs.c

```diff
@@ -4688,43 +4688,85 @@ static int64_t get_blkdev_size(struct stat *st)
 #endif
 }
 
+#ifndef __CYGWIN__
+/*!
+ *****************************************************************************
+ *
+ ****************************************************************************/
+static int parse_fuse_fd(const char *mountpoint)
+{
+        int fd = -1;
+        unsigned int len = 0;
+
+        if (sscanf(mountpoint, "/dev/fd/%u%n", &fd, &len) == 1 && len == strlen(mountpoint)) {
+                return fd;
+        }
+
+        return -1;
+}
+#endif
+
 /*!
  *****************************************************************************
  *
  ****************************************************************************/
 static int check_paths(const char *prog, char *src_path_in, char *dst_path_in,
                 char **src_path_out, char **dst_path_out, int verbose)
 {
+        struct stat st;
+
         char *a1 = realpath(src_path_in, NULL);
+
+        if (!a1) {
+                if (verbose)
+                        printf("%s: invalid source: %s\n", prog, src_path_in);
+                return -1;
+        }
+
 #ifdef __CYGWIN__
         char *a2 = dst_path_in;
 #else
-        char *a2 = realpath(dst_path_in, NULL);
+        char *a2;
+
+        /* Check if destination path is a pre-mounted FUSE descriptor. */
+        const int fuse_fd = parse_fuse_fd(dst_path_in);
+        if (fuse_fd >= 0) {
+                if (verbose)
+                        printf("%s: using pre-mounted FUSE descriptor %u\n", prog, fuse_fd);
```

Comment:
I suggest you syslog this type of information instead. The verbose flag was/is intended for failures, not success.

Reply:
Removed. This was useful to me for debugging, but shouldn't stay in production.